pages/login.js

```diff
@@ -0,0 +1,79 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import * as crypto from "crypto";
+import styles from "../styles/RegisterAndLogin.module.css";
+
+export default function Login() {
+  const router = useRouter();
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const [token, setToken] = useState("");
+
+  useEffect(() => {
+    setToken(window.localStorage.getItem("token"));
+  }, []);
+
+  function clearToken() {
+    window.localStorage.setItem("token", "");
+    console.log(token);
+    router.reload();
+  }
+
+  function validateUser(event) {
+    event.preventDefault();
+    const user = JSON.parse(window.localStorage.getItem("user"));
+
+    const checkUserInputDataWithLocalStorageUserData =
+      user.email === email && user.password === password;
+
+    if (checkUserInputDataWithLocalStorageUserData) {
+      const token = crypto.createHash("sha256").update(password).digest("hex");
+      window.localStorage.setItem("token", token);
+      router.reload();
```

Comment:
Instead of reloading route you should change a state which stores a boolean value if user is logged in or not. And then depending on that render a different HTML.
When React updates state it will do an internal re-render.